dymos/utils/introspection.py

```diff
@@ -1199,95 +1138,74 @@ def _get_targets_metadata(ode, name, user_targets=_unspecified, user_units=_unsp
-    else:
-        units = user_units
-
-    # Resolve whether the targets is static or dynamic
-    static_target_tags = [tgt for tgt in targets if 'dymos.static_target' in ode_inputs[tgt]['tags']]
-    if static_target_tags:
-        static_target = True
-        if not user_static_target:
-            raise ValueError(f"User has specified 'static_target = False' for parameter {name},"
-                             f"but one or more targets is tagged with "
-                             f"'dymos.static_target': {' '.join(static_target_tags)}")
-    else:
-        if user_static_target is _unspecified:
-            static_target = False
-        else:
-            static_target = user_static_target
+    for tgt, options in targets.items():
+        options['units'] = ode_inputs[tgt]['units']
+        options['shape'] = ode_inputs[tgt]['shape']
+        options['val'] = ode_inputs[tgt]['val']
+        options['tags'] = ode_inputs[tgt]['tags']
+
+    return targets
 
-    if user_shape in {None, _unspecified}:
-        # Resolve target shape
-        target_shape_set = {ode_inputs[tgt]['shape'] for tgt in targets}
-        if len(target_shape_set) == 1:
-            shape = next(iter(target_shape_set))
-            if not static_target:
-                if len(shape) == 1:
-                    shape = (1,)
-                else:
-                    shape = shape[1:]
-        elif len(target_shape_set) == 0:
-            raise ValueError(f'Unable to automatically assign a shape to {name}.\n'
-                             'Targets for this variable either do not exist or have no shape set.\n'
-                             'The shape for this variable must be set explicitly via the '
-                             '`shape=<tuple>` argument.')
-        else:
-            raise ValueError(f'Unable to automatically assign a shape to {name} based on targets. '
-                             f'Targets have multiple shapes assigned: {target_shape_set}. '
-                             f'Change targets such that all have common shapes.')
-    else:
-        shape = user_shape
 
-    return targets, shape, units, static_target
+def _get_common_metadata(targets, metadata_key):
+    """
+    Given a dictionary containing targets and their metadata, return the value associated
+    with the given metadata key if that value is common to all targets, otherwise raise an
+    Exception.
+
+    Parameters
+    ----------
+    targets : dict
+        A dictionary of targets and their metadata which must include the desired metadata key.
+    metadata_key : str
+        The metadata key desired.
+
+    Returns
+    -------
+    object
+        The common metadata value shared by all targets.
+
+    Raises
+    ------
+    ValueError
+        ValueError is raised if the targets do not all have the same metadata value.
+    """
+    meta_set = {meta[metadata_key] for tgt, meta in targets.items()}
```

Comment:
You only need the metadata here, not the target name, so could just use `targets.values()`